dpe/src/dpe_instance.rs

```diff
@@ -472,6 +480,35 @@ impl<C: Crypto> DpeInstance<'_, C> {
     }
 }
 
+struct BufWriter<'a> {
+    buf: &'a mut [u8],
+    offset: usize,
+}
+
+impl Write for BufWriter<'_> {
+    fn write_str(&mut self, s: &str) -> Result<(), Error> {
+        if s.len() > self.buf.len().saturating_sub(self.offset) {
+            return Err(Error::default());
+        }
+
+        self.buf[self.offset..self.offset + s.len()].copy_from_slice(s.as_bytes());
+        self.offset += s.len();
+
+        Ok(())
+    }
+}
+
+impl BufWriter<'_> {
+    fn write_hex_str(&mut self, src: &[u8]) -> Result<(), DpeErrorCode> {
+        for &b in src {
+            self.write_fmt(format_args!("{b:02x}"))
+                .map_err(|_| DpeErrorCode::InternalError)?;
```

Comment:
optional: this could be simplified a little
```suggestion
            write!(self, "{b:02x}").map_err(|_| DpeErrorCode::InternalError)?;
```

Reply:
Done